File: ObserverPattern/app/src/main/java/com/usu/observerpattern/MainActivity.java
package com.usu.observerpattern;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {
    Count count = new Count();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    public Count getCount() {
        return count;
    }

    public void increment() {
        count.increment();
    }

    public void decrement() {
        count.decrement();
    }
}

File: AnActualKotlinApplication/app/src/main/java/com/usu/anactualkotlinapplication/MainActivity.kt
package com.usu.anactualkotlinapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModelProvider

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val viewModel: MainViewModel = ViewModelProvider(this).get(MainViewModel::class.java)
        val countTextView: TextView = findViewById(R.id.count);
        viewModel.count = MutableLiveData();
        viewModel.count.observe(this) {
            countTextView.text = "$it"
        }
        val incrementButton: Button = findViewById(R.id.increment)
        incrementButton.setOnClickListener {
            viewModel.increment()
        }
        val decrementButton: Button = findViewById(R.id.decrement);
        decrementButton.setOnClickListener {
            viewModel.decrement()
        }
    }
}